scala-generator/src/main/scala/models/http4s/server/Route.scala

```diff
@@ -19,7 +19,7 @@ case class Route(ssd: ScalaService, resource: ScalaResource, op: ScalaOperation,
     if(!segment.startsWith(":")) {
       Literal(segment)
     } else {
-      val truncated = segment.drop(1)
+      val truncated = ScalaUtil.toVariable(segment.drop(1))
```

Comment:
I think we need to look at the original parameter name as it appears on the path, rather than the scala variable form of it. Instead of converting the path segment into a scala variable, we could simply compare the truncated segment to the parameters original name in the line below. Ie `find(_.originalName == truncated)`

Reply:
I think the variable name is what needs to be used in the route matcher b/c the variable name is used in the function call below it.
i.e.
```
case _req @ GET -> Root / "named_enum" / EnumVal(namedId) if apiVersionMatch(_req) =>
      getNamedEnumByNamedId(_req, namedId).flatMap {
        case GetNamedEnumByNamedIdResponse.HTTP200(headers) => Ok().putHeaders(headers: _*)
      }
```
```
def route(version: Option[Int]): Seq[String] = {
    val args = (
      Seq(Some("_req")) ++
        op.nonHeaderParameters.map(field => Some(s"${ScalaUtil.quoteNameIfKeyword(field.name)}")) ++
        Seq(op.body.map(body => s"_req.attemptAs[${body.datatype.name}]"))
      ).flatten.mkString(", ")
```

Reply:
So we can check the path parameters against the originalName but then the `truncated` variable needs to be converted to a paramName two lines later
```
case ScalaPrimitive.String if scalaParameter.param.minimum.isEmpty && scalaParameter.param.maximum.isEmpty => PlainString(truncated)
          case _: ScalaPrimitive => Extracted(truncated, scalaParameter)
```

Reply:
If you look at the implementation of ScalaParameter.name, it does `ScalaUtil.toVariable(param.name)` while `def originalName = param.name`. So we can either call ScalaUtil.toVariable on the extracted path segment as you do, or we can compare the "unadulterated" path segment with the "unadulterated" parameter name. I know it's an edge case (and ultimately shouldn't be supported by apibuilder), but you could have two path params, one `ord_id` and the other `ordId`, and a path `/foo/:ord_id/bar/:orgId`. After `ScalaUtil.toVariable` both will be `ordId` which then makes it impossible to figure out which of the two parameters is being referred to. If you look up by the original name ("org_id" or "orgId") we know which parameter they respond to.

Reply:
And sorry for being pedantic :)